test/nodes/test_question_generator.py

```diff
@@ -22,6 +83,47 @@ def test_qg_pipeline(question_generator):
     assert len(result["generated_questions"][0]["questions"]) > 0
 
 
+def test_qg_pipeline_non_default_params():
+    question_generator = QuestionGenerator(model_name_or_path="valhalla/t5-small-e2e-qg", num_queries_per_doc=2)
+    p = QuestionGenerationPipeline(question_generator)
+    result = p.run(documents=[document, document_2])
+    assert isinstance(result, dict)
+    assert "generated_questions" in result
+    assert "documents" in result
+    assert isinstance(result["generated_questions"], list)
+    assert isinstance(result["documents"], list)
+    assert len(result["generated_questions"]) == 2
+    assert len(result["documents"]) == 2
+    assert len(result["generated_questions"][0]["questions"]) > 0
+    assert len(result["generated_questions"][1]["questions"]) > 0
```

Comment:
Would it be possible to replace `> 0` with `== num`? This would have caught the `min` vs. `max` in the init function.

Reply:
Hmm, we could; this number would be the number of splits * num_queries_per_doc. I was afraid to make the test too brittle but I think we can do this.

Reply:
I think it would be a good idea. Since the number should be `number of splits * num_queries_per_doc` I don't think this number would need to be updated unless the test document changed and then the keyword testing might also need to be changed.

Reply:
Ok, done, thanks @sjrl this way we tighten up the test requirements even further but that's ok.